pkg/cfn/manager/api.go

```diff
@@ -449,15 +475,22 @@ func (c *StackCollection) DescribeStacks() ([]*Stack, error) {
 	}
 	return stacks, nil
 }
-
 func (c *StackCollection) IsClusterStack() (bool, error) {
-	clusterStackName := c.makeClusterStackName()
-	stacks, err := c.DescribeStacks()
+	clusterStackNames, err := c.ListStackNamesMatching("eksctl-.*-cluster")
 	if err != nil {
 		return false, err
 	}
-	for _, stack := range stacks {
-		if *stack.StackName == clusterStackName {
+	return c.IsClusterStackUsingCachedList(clusterStackNames)
+}
+
+func (c *StackCollection) IsClusterStackUsingCachedList(clusterStackNames []string) (bool, error) {
+	clusterStackName := c.makeClusterStackName()
+	for _, stack := range clusterStackNames {
+		if stack == clusterStackName {
+			stack, err := c.DescribeStack(&cloudformation.Stack{StackName: &clusterStackName})
+			if err != nil {
+				return false, err
+			}
 			for _, tag := range stack.Tags {
 				if matchesClusterName(*tag.Key, *tag.Value, c.spec.Metadata.Name) {
 					return true, nil
```

Comment:
We could remove the tag check entirely to remove all `DescribeStack` calls, but I think its a check worth doing. LMK if anyone disagrees